lib/plugins/Label_4N.ts

```diff
@@ -0,0 +1,75 @@
+import { DecoderPlugin } from '../DecoderPlugin';
+import { DecodeResult, Message, Options } from '../DecoderPluginInterface';
+import { CoordinateUtils } from '../utils/coordinate_utils';
+import { ResultFormatter } from '../utils/result_formatter';
+
+export class Label_4N extends DecoderPlugin {
+  name = 'label-4n';
+
+  qualifiers() { // eslint-disable-line class-methods-use-this
+    return {
+      labels: ['4N'],
+    };
+  }
+
+  decode(message: Message, options: Options = {}) : DecodeResult {
+    const decodeResult = this.defaultResult();
+    decodeResult.decoder.name = this.name;
+    decodeResult.message = message;
+    decodeResult.formatted.description = 'Airline Defined';
+
+    // Inmarsat C-band seems to prefix normal messages with a message number and flight number
+    let text = message.text;
+    if (text.match(/^M\d{2}A\w{6}/)) {
+        ResultFormatter.flightNumber(decodeResult, message.text.substring(4, 10).replace(/^([A-Z]+)0*/g, "$1"));
+        text = text.substring(10);
+    }
+
+    decodeResult.decoded = true;
+    const fields = text.split(",");
+    if (text.length === 51) {
+        // variant 1
+        decodeResult.raw.day_of_month = text.substring(0, 2);
+        ResultFormatter.departureAirport(decodeResult, text.substring(8, 11));
+        ResultFormatter.arrivalAirport(decodeResult, text.substring(13, 16));
+        ResultFormatter.position(decodeResult, CoordinateUtils.decodeStringCoordinatesDecimalMinutes(text.substring(30, 45).replace(/^(.)0/, "$1")));
+        ResultFormatter.altitude(decodeResult, text.substring(48, 51) * 100);
+        decodeResult.remaining.text = [text.substring(2, 4), text.substring(19, 29)].join(" ");
+    } else if (fields.length === 33) {
+        // variant 2
+        decodeResult.raw.date = fields[3];
+        if (fields[1] === "B") {
+            ResultFormatter.position(decodeResult, {latitude: Number(fields[4]), longitude: Number(fields[5])});
+            ResultFormatter.altitude(decodeResult, fields[6]);
+        }
+        ResultFormatter.departureAirport(decodeResult, fields[8]);
+        ResultFormatter.arrivalAirport(decodeResult, fields[9]);
```

Comment:
idx 10 is alternate airport and 12 is the alternate runway. I made a note of this in Label 10. maybe it's worth parsing.

Reply:
added, you might want changes in the result_formatter bit